### PR TITLE
DOC: PDEP6 page format update

### LINKED ISSUE #53414: DOC: PDEP6 page doesn't render correctly
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/pdeps/0006-ban-upcasting.html

### Documentation problem

here, for example, it should be a bullet list:

image

### Suggested fix for documentation

fixup formatting

you may want to build the page locally first, please check https://github.com/pandas-dev/pandas/blob/main/web/README.md for how to do that, and the contributing guide https://pandas.pydata.org/docs/development/contributing.html for how to set up your environment

### PR DESCRIPTION
**Pull request by UmbertoFasci**
- [x] closes #53414
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by UmbertoFasci** [comment]
Here is link to preview the change:

https://pandas.pydata.org/preview/53442/pdeps/0006-ban-upcasting.html

**Comment by UmbertoFasci** [comment]
#debf592 updates the entire document format where bulleted lists are suggested.

File: web/pandas/pdeps/0006-ban-upcasting.md [hunk 1]
```diff
@@ -75,18 +75,20 @@ It may be desirable to expand the top list to ``Series.replace`` and ``Series.up
...
+- ``df['a'] = pd.date_range(datetime(2020, 1, 1), periods=3)``
+- ``df.index.intersection(ser.index)``
 
 ## Detailed description
 
 Concretely, the suggestion is:
-- if a ``Series`` is of a given dtype, then a ``setitem``-like operation should not change its dtype;
+
+- if a ``Series`` is of a given dtype, then a ``setitem``-like operation should not change its dtype.
 - if a ``setitem``-like operation would previously have changed a ``Series``' dtype, it would now raise.
```
Comment (UmbertoFasci) [inline web/pandas/pdeps/0006-ban-upcasting.md:92]:
```markdown
``Series``' 
```
Is this an intentional formatting?

**Comment by UmbertoFasci** [comment]
Pre-commit failure found in [7722182](https://github.com/pandas-dev/pandas/pull/53442/commits/772218239ffd081d2e6a421838021edc76722763) and [76a687f](https://github.com/pandas-dev/pandas/pull/53442/commits/76a687f02f9e9267da0a7e12beeefa443d259635) started to occur after performing local build. I will address this ASAP.

**Review by MarcoGorelli — APPROVED** [review]
mostly looks really good, thanks @UmbertoFasci !

Just a couple of places don't look quite right:

image

image

File: web/pandas/pdeps/0006-ban-upcasting.md [hunk 2]
```diff
@@ -51,65 +51,71 @@ In[11]: ser[2] = "2000-01-04x"  # typo - but pandas does not error, it upcasts t
...
+    ```python
+    value = extract_array(value, extract_numpy=True)
+    try:
+        casted = np_can_hold_element(values.dtype, value)
+    except LossySetitemError:
+        # current dtype cannot store value, coerce to common dtype
+        nb = self.coerce_to_target_dtype(value)
+        return nb.setitem(indexer, value)
+    else:
+    ```
```
Comment (MarcoGorelli) [inline web/pandas/pdeps/0006-ban-upcasting.md:110]: this one may not be rendering correctly

File: web/pandas/pdeps/0006-ban-upcasting.md [hunk 3]
```diff
@@ -216,13 +225,15 @@ at all. To keep this proposal focused, it is intentionally excluded from the sco
...
+**A**: You may want to define your own helper function, such as:
+
   ```python
-  >>> def maybe_convert_to_int(x: int | float, tolerance: float):
+  def maybe_convert_to_int(x: int | float, tolerance: float):
       if np.abs(x - round(x)) < tolerance:
           return round(x)
       return x
   ```
+
```
Comment (MarcoGorelli) [inline web/pandas/pdeps/0006-ban-upcasting.md:236]: likewise

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: web/pandas/pdeps/0006-ban-upcasting.md (hunk 3 above)
Comment (UmbertoFasci) [reply web/pandas/pdeps/0006-ban-upcasting.md]: The FAQ source code block should render correctly now, and I will resolve upon full code check and preview pass.

Comment (MarcoGorelli) [reply web/pandas/pdeps/0006-ban-upcasting.md]: sure, thanks - just fyi there's no need to merge upstream/main unless you have to (e.g. if there's conflicts, or if it's an old PR)

Comment (UmbertoFasci) [reply web/pandas/pdeps/0006-ban-upcasting.md]: Local build has indicated proper code block formating for this code block in the FAQ.

File: web/pandas/pdeps/0006-ban-upcasting.md (hunk 2 above)
Comment (UmbertoFasci) [reply web/pandas/pdeps/0006-ban-upcasting.md]: As of [e617561](https://github.com/pandas-dev/pandas/pull/53442/commits/e6175619d185c7548f30f556f09f828855c9314b) the code block within the ordered list is rendered correctly and does not break the ordered list according to local build. Code check in progress.

**Comment by UmbertoFasci** [comment]
Thank you @MarcoGorelli, working on this issue allowed me to learn a lot. I will use pre-commit in my next PR workflow to handle those trailing whitespaces that I seem to love making.

**Review by MarcoGorelli — APPROVED** [review]
nice one, thanks @UmbertoFasci

**Pull request merged**